### ISSUE TITLE
Compile error re: TimeUtils.swift

### ISSUE THREAD

**Issue by stonetip** [apple/swift-protobuf#341]
Using 0.9.26 as installed from podfile () I ran into a compile error in Xcode 8.2. In the timeOfDayFromSecondsSince1970 func, I had to optionally cast hh to Int32. (Expedient fix just to continue compiling and testing).

```
    let hh = div(secondsSinceMidnight, secondsPerHour) as? Int32
    
    return (hh: hh!, mm: mm, ss: ss)
```

**Comment by tbkka** [comment]
@cburrows:  Could you take a look at this?

**Comment by thomasvl** [comment]
Given this was 0.9.26, do we want to worry vs. just let it get fixed when are ready to cut a new release?

**Comment by tbkka** [comment]
@stonetip:  I just tagged the current tip-of-tree as 0.9.28.  Could you try that version to see if you still get this error?  I'm confused why no-one else is seeing this, since I don't think anything there has changed since 0.9.26.

**Comment by stonetip** [comment]
I haven't yet tried 0.9.28, However, I copied the function and constants over from TimeUtils.swift and the mod and div functions from MathUtils.swift to a playground and ran a test against the current epoch time. Of course, no problems there!

Oddly, earlier when tried "Jump to Definition" on **div** it was opening a Darwin class. I came back to the computer a few hours later and was taken to MathUtils.swift instead. I don't know if this has any bearing on the issue.

Regardless, I have to either modify the hh line to:

`let hh = div(secondsSinceMidnight, secondsPerHour) as Int32`

or 

`let hh = Int32(div(secondsSinceMidnight, secondsPerHour))`

to compile a project targeting iOS 10.2 in Xcode 8.2.1 (8C1002)
